source/extensions/resource_monitors/envoy_container_cpu_utilization/envoy_container_cpu_utilization_monitor.cc

```diff
@@ -0,0 +1,76 @@
+#include "source/extensions/resource_monitors/envoy_container_cpu_utilization/envoy_container_cpu_utilization_monitor.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <thread>
+
+#include "source/common/common/fmt.h"
+
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace EnvoyContainerCpuUtilizationMonitor {
+
+// The dampening alpha value used for EWMA calculation.
+// The value is chosen to be very small to give past calculations higher priority.
+// This helps in reducing the impact of sudden spikes or drops in CPU utilization.
+constexpr double DAMPENING_ALPHA = 0.05;
+
+EnvoyContainerCpuUtilizationMonitor::EnvoyContainerCpuUtilizationMonitor(
+    const envoy::extensions::resource_monitors::envoy_container_cpu_utilization::v3::
+        EnvoyContainerCpuUtilizationConfig& /*config*/,
+    std::unique_ptr<ContainerStatsReader> envoy_container_stats_reader)
+    : envoy_container_stats_reader_(std::move(envoy_container_stats_reader)) {
+  previous_envoy_container_stats_ = envoy_container_stats_reader_->getEnvoyContainerStats();
+}
+
+void EnvoyContainerCpuUtilizationMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks& callbacks) {
```

Comment:
If we did the folding this file would effectively merge with the existing https://github.com/envoyproxy/envoy/blob/main/source/extensions/resource_monitors/cpu_utilization/cpu_utilization_monitor.h which it looks to me to be very similar.